uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkDay.php

```diff
@@ -0,0 +1,124 @@
+
+  /**
+   * Create a new WorkDay based on array-data
+   *
+   * @param array $params key-value pairs
+   * @return CRM_HRLeaveAndAbsences_DAO_WorkDay|NULL
+   *
+   */
+  public static function create($params) {
+    $className = 'CRM_HRLeaveAndAbsences_DAO_WorkDay';
+    $entityName = 'WorkDay';
+    $hook = empty($params['id']) ? 'create' : 'edit';
+    self::validateParams($params);
+
+    if(!empty($params['id'])) {
+      unset($params['week_id']);
+    }
+
+    CRM_Utils_Hook::pre($hook, $entityName, CRM_Utils_Array::value('id', $params), $params);
+    $instance = new $className();
+    $instance->copyValues($params);
+    $instance->save();
+    CRM_Utils_Hook::post($hook, $entityName, $instance->id, $instance);
+
+    return $instance;
+  }
+
+  /**
+   * Return a list of possible options for the WorkDay::type field.
+   *
+   * The list is the format $value => $label, so it can be used
+   * to populate select controls.
+   *
+   * @return array the list of possible options
+   */
+  public static function getWorkTypeOptions()
+  {
+    return [
+      self::WORK_DAY_OPTION_NO => ts('No'),
+      self::WORK_DAY_OPTION_YES => ts('Yes'),
+      self::WORK_DAY_OPTION_WEEKEND => ts('Weekend'),
+    ];
+  }
+
+  private static function validateParams($params) {
+    self::validateWorkDayType($params);
+    self::validateDayOfTheWeek($params);
+    self::validateWorkHours($params);
+  }
+
+  private static function validateDayOfTheWeek($params)
+  {
+    $dayOfTheWeek = empty($params['day_of_the_week']) ? null : $params['day_of_the_week'];
+
+    if(!is_int($dayOfTheWeek) || ($dayOfTheWeek < 1 || $dayOfTheWeek > 7)) {
+      throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException(
+        'Day of the Week should be a number between 1 and 7, according to ISO-8601'
+      );
+    }
+  }
+
+  private static function validateWorkHours($params)
+  {
+    $typeOfDay = empty($params['type']) ? null : $params['type'];
+    $timeFrom = empty($params['time_from']) ? null : $params['time_from'];
+    $timeTo = empty($params['time_to']) ? null : $params['time_to'];
+    $break = empty($params['break']) ? null : $params['break'];
+
+    if($typeOfDay != self::WORK_DAY_OPTION_YES && ($timeFrom || $timeTo || $break)) {
+      throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException(
+        'Time From, Time To and Break should be empty for Non Working Days and Weekends'
+      );
+    }
+
+    if($timeFrom && !preg_match('/^([01][0-9]|2[0-3]):[0-5][0-9]$/', $timeFrom)) {
+      throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException(
+        'Time From format should be hh:mm'
+      );
+    }
+
+    if($timeTo && !preg_match('/^([01][0-9]|2[0-3]):[0-5][0-9]$/', $timeTo)) {
+      throw new CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException(
+        'Time To format should be hh:mm'
+      );
+    }
+
+    if($typeOfDay == self::WORK_DAY_OPTION_YES && !($timeFrom && $timeTo && $break)) {
```

Comment:
Please can you enclose statements like these in (), so for example this 
`$typeOfDay == self::WORK_DAY_OPTION_YES && !($timeFrom && $timeTo && $break)`
could be
`$typeOfDay == (self::WORK_DAY_OPTION_YES && !($timeFrom && $timeTo && $break))`. 
The reason is because there are multiple logical operators in these statements and enclosing would make it more readable. Thanks!